mlx_graphs/data/collate.py

```diff
@@ -0,0 +1,51 @@
+from typing import List
+
+import mlx.core as mx
+from mlx_graphs.data.data import GraphData
+
+
+def collate(graph_list: List[GraphData]) -> dict:
+    """
+    Collate function to perform the unification of graphs in a batch.
+    By default, it concatenates all default graph attributes in dim 0
+    apart from `edge_index` which is concatenated along dim 1 and
+    nodes indices are incremented
+
+    Args:
+        graph_list (List[GraphData]): the list of GraphData objects to collate
+
+    Returns:
+        dict: dict containing all the attributes of the unified and disconnected big graph as
+        well as the slices corresponding to the portion of node for each graph from the provided
+        list of graphs.
```

Comment:
No need for type specs in docstrings as they're now inferred from the signature

```suggestion
    Args:
        graph_list: the list of GraphData objects to collate

    Returns:
        Dict containing all the attributes of the unified and disconnected big graph as
        well as the slices corresponding to the portion of node for each graph from the provided
        list of graphs.
```